src/workerd/api/http.c++

```diff
@@ -161,14 +161,14 @@ Headers::Headers(jsg::Lock& js, const Headers& other): guard(Guard::NONE) {
 }
 
 Headers::Headers(jsg::Lock& js, const kj::HttpHeaders& other, Guard guard): guard(Guard::NONE) {
-  other.forEach([this](kj::StringPtr name, kj::StringPtr value) {
-    // The variation of toLower we use here creates a copy.
-    auto key = jsg::ByteString(toLower(name));
-    KJ_IF_SOME(header, headers.find(key)) {
-      header.values.add(jsg::ByteString(kj::str(value)));
+  other.forEachHeader([this](kj::HttpHeaders::Header header) {
+    auto key = jsg::ByteString(kj::str(header.lowcaseName));
```

Comment:
It would be ideal if we could avoid the additional allocation (`kj::str(...)`) and wrapping it in `jsg::ByteString` here.

Reply:
Byte string is a kj::string, and kj::string constructor just moves the value. So technically there is no unnecessary allocation here.